flake.nix

```diff
@@ -1,88 +1,58 @@
 {
-
   inputs = {
     nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
-    flake-parts.url = "github:hercules-ci/flake-parts";
     rust-overlay.url = "github:oxalica/rust-overlay";
+    rust-overlay.inputs.nixpkgs.follows = "nixpkgs";
   };
 
-  outputs = inputs@{ nixpkgs, flake-parts, rust-overlay, ... }:
-    flake-parts.lib.mkFlake {
-      inherit inputs;
-    } {
-      systems = [
+  outputs = { self, nixpkgs, rust-overlay, ... }:
+    let
+      forAllSystems = nixpkgs.lib.genAttrs [
+        "aarch64-linux"
```

Comment:
fyi we do not have any testing setup for either arm or 32-bit and i dont think we have plans to test on it for now

Reply:
Well, it should work for arm and 32-bit. I don't believe magmawm has any architecture dependent code. Idk tho

Reply:
it *should* but we have no real way of testing it as far as i know

Reply:
> it *should* but we have no real way of testing it as far as i know

Anyone with an apple silicon device with asahi linux on it feel free to hit me up to get this tested, actually now that i wrote this i realised we could just test on a raspberry pi aswell

Reply:
makes sense

I'm unsure if we should even bother with 32 bit, most computers nowadays (even the low end ones) are 64 bit as well